packages/cli/src/tokens/build/utils/getMultidimensionalThemes.ts

```diff
@@ -0,0 +1,153 @@
+import type { ThemeObject } from '@tokens-studio/types';
+import { TokenSetStatus } from '@tokens-studio/types';
+import chalk from 'chalk';
+import { camelCase } from 'change-case';
+import * as R from 'ramda';
+import { buildOptions } from '../../build';
+import type { ThemeDimension, ThemePermutation } from '../types';
+
+/**
+ * Find the theme permutations that are relevant for the given theme dimensions.
+ *
+ * Technically, for the given dimensions all permutations are included, while for other
+ * dimensions the first permutation is used.
+ *
+ * @param themes Theme objects from $themes.json (Tokens Studio)
+ * @param dimensions Which theme dimensions to return permutations for.
+ *    'theme' (e.g. altinn/digdir/uutilsynet) is always implicitly included.
+ * @returns the relevant theme permutations
+ */
+export const getMultidimensionalThemes = (themes: ThemeObject[], dimensions: ThemeDimension[]) => {
+  const verboseLogging = buildOptions?.verbose;
+  const grouped$themes = groupThemes(themes);
+  const permutations = permutateThemes(grouped$themes);
+  const ALL_DEPENDENT_ON: ThemeDimension[] = ['theme'];
+  const keys = R.keys(grouped$themes);
+  const nonDependentKeys = keys.filter((x) => ![...ALL_DEPENDENT_ON, ...dimensions].includes(x));
+  if (verboseLogging) {
+    console.log(chalk.cyan(`🔎 Finding theme permutations for ${dimensions}`));
+    console.log(chalk.cyan(`   (ignoring permutations for ${nonDependentKeys})`));
+  }
+  return permutations.filter((val: PermutatedTheme) => {
+    const filters = nonDependentKeys.map((x) => val[x] === grouped$themes[x][0].name);
+    return filters.every((x) => x);
+  });
+};
+
+export type PermutatedTheme = {
+  name: string;
+  selectedTokenSets: string[];
+} & ThemePermutation;
+
+const processed: unique symbol = Symbol('Type brand for ProcessedThemeObject');
+type ProcessedThemeObject = ThemeObject & { [processed]: true };
+function isProcessed(theme: ThemeObject | ProcessedThemeObject): theme is ProcessedThemeObject {
+  return Boolean((theme as ProcessedThemeObject)[processed]);
+}
+
+/**
+ * Normalise theme names and theme group names for easier use in code
+ * @param theme A theme object from $themes.json
+ * @returns Processed theme object
+ */
+function processThemeObject(theme: ThemeObject | ProcessedThemeObject): ProcessedThemeObject {
+  if (isProcessed(theme)) {
+    return theme;
+  }
+  const result: ProcessedThemeObject = { ...theme, [processed]: true };
+  if (result.group) {
+    result.group = camelCase(result.group);
```

Comment:
I suggest we use the same casing for `name` and `group` on our themes so its more predictable to use in code. Ideally lowerCase, but maybe `noCase` since we might have group names with spaces?

Reply:
What does `noCase` actually do? I like `camelCase` since it is quite predictable:
```ts
import camelCase from 'camelcase';

camelCase('foo-bar');
//=> 'fooBar'

camelCase('foo_bar');
//=> 'fooBar'

camelCase('Foo-Bar');
//=> 'fooBar'

camelCase('--foo.bar');
//=> 'fooBar'

camelCase('foo bar');
//=> 'fooBar'

camelCase('Foo bar');
//=> 'fooBar'

camelCase(['__foo__', '--bar']);
//=> 'fooBar'
```

Reply:
`noCase` is just `kebab-case` with some additional parsing for replacing spaces. Maybe we should rename it then? 

Its something thats been there since before I started and we still had some design-tokens we needed it for.

I think whats most important is we use the same case conversion for our fields for further use. Since we've already started using kebab-case for folders in design-tokens and names I feel its maybe the best fit?

Reply:
Ok, I haven't looked that the code for new color-modes so wasn't aware this was made for that. Not sure I understand at this point why we need different formatters for main/support but lets save that for later :) 

But lets user `kebab-case` for `name` and `camelCase` for `group` for now and we'll see where the rest falls, once #2507 and #2670 are solved.

Reply:
Yeah, I'll just set it to `kebab-case` for both right now since it doesn't affect anything else in this PR, then I'll see how it works out down the line!